src/components/Folder/Folder.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from "react";
+import { useGetFolderById } from "../../hooks/useGetFolder";
+import SearchBar from "../SearchBar/SearchBar";
+import "./Folder.css";
+import SelectedOption from "../SelectedOption/SelectedOption";
+import FolderList from "../FolderList/FolderList";
+
+function Folder() {
+  const { data } = useGetFolderById();
+  const dataArray = data?.data;
+
+  const [activeButton, setActiveButton] = useState("전체");
+  const [selectedFolder, setSelectedFolder] = useState("");
+
+  useEffect(() => {
+    if (selectedFolder === "" && dataArray && dataArray.length > 0) {
+      setSelectedFolder("전체");
+    }
+  }, [selectedFolder, dataArray]);
+
+  const handleButtonClick = (name) => {
+    setActiveButton(name);
+    setSelectedFolder(name);
+  };
+
+  let renderedButtons;
+
+  if (dataArray && Array.isArray(dataArray)) {
+    renderedButtons = dataArray.map((item) => (
+      <button
+        key={item.id}
+        className={activeButton === item.name ? "active" : ""}
+        onClick={() => handleButtonClick(item.name)}
+      >
+        {item.name}
+      </button>
+    ));
```

Comment:
컴포넌트 중간에 이런식으로 코드 전개는 잘 하지 않는 편인 것 같아요.
함수로 만들어주셔서 아래에서 함수 실행을 해줘도 괜찮을 것 같습니다!

```suggestion

const renderButton = () => {
  if (dataArray && Array.isArray(dataArray)) {
    return dataArray.map((item) => (
      <button
        key={item.id}
        className={activeButton === item.name ? "active" : ""}
        onClick={() => handleButtonClick(item.name)}
      >
        {item.name}
      </button>
    ));
    
    return null;
 }
```